cobalt/browser/metrics/cobalt_metrics_service_client.cc

```diff
@@ -41,66 +38,41 @@ class CobaltMetricsLogUploader : public metrics::MetricsLogUploader {
                  const std::string& log_hash,
                  const std::string& log_signature,
                  const metrics::ReportingInfo& reporting_info) {
-    // TODO(b/372559349): Add logic here to report the Blob to the WebApp.
-    NOTIMPLEMENTED();
+    DLOG(INFO) << "UMA Payload uploading! Hash: " << log_hash;
   }
 };
 
 }  // namespace
 
-CobaltMetricsServiceClient::CobaltMetricsServiceClient()
-    : synthetic_trial_registry_(new variations::SyntheticTrialRegistry) {
-  DETACH_FROM_THREAD(thread_checker_);
-
-  // TODO(b/372559349): Extract initialization into a method if any of the
-  // following calls can fail (e.g. base::PathService::Get()).
-
-  // No need to make `pref_registry` a member, `pref_service_` will keep a
-  // reference to it.
-  auto pref_registry = base::MakeRefCounted<PrefRegistrySimple>();
-  metrics::MetricsService::RegisterPrefs(pref_registry.get());
-
-  PrefServiceFactory pref_service_factory;
-  // TODO(b/397929564): Investigate using a Chrome's memory-mapped file store
-  // instead of in-memory.
-  pref_service_factory.set_user_prefs(
-      base::MakeRefCounted<InMemoryPrefStore>());
-
-  pref_service_ = pref_service_factory.Create(std::move(pref_registry));
-  CHECK(pref_service_);
-  // `local_state` is a common alias used in Content embedders.
-  auto* local_state = pref_service_.get();
-
-  base::FilePath user_data_dir;
-  // TODO(b/372559349): use a real path.
-  base::PathService::Get(content::PATH_START, &user_data_dir);
-
-  metrics_state_manager_ = metrics::MetricsStateManager::Create(
-      local_state, /* enabled_state_provider= */ this,
-      /* backup_registry_key= */ std::wstring(), user_data_dir
-      // Other params left as by-default.
-  );
-  metrics_state_manager_->InstantiateFieldTrialList();
-
-  metrics_service_ = std::make_unique<metrics::MetricsService>(
-      metrics_state_manager_.get(), this, pref_service_.get());
-  metrics_service_->InitializeMetricsRecordingState();
-}
-
-CobaltMetricsServiceClient::~CobaltMetricsServiceClient() {
-  Stop();
+CobaltMetricsServiceClient::CobaltMetricsServiceClient(
+    metrics::MetricsStateManager* state_manager,
+    variations::SyntheticTrialRegistry* synthetic_trial_registry,
+    PrefService* local_state)
+    : metrics_state_manager_(state_manager),
+      synthetic_trial_registry_(synthetic_trial_registry),
+      local_state_(local_state) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
```

Comment:
I think this one should be DETACH_FROM_THREAD as it was previously

Reply:
Done